syntax = "proto3";

package file;

service FileService {
    rpc SayHello (HelloRequest) returns (HelloResponse) {}
    rpc GetServerTime (Empty) returns (TimeResponse) {}
    rpc WriteFile (FileRequest) returns (FileResponse) {}
    rpc ReadFile (FileRequest) returns (FileContentResponse) {}
    rpc DeleteFile (FileRequest) returns (FileResponse) {}
    rpc GetFilesList (Empty) returns (FilesListResponse) {}
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string message = 1;
}

message Empty {}

message TimeResponse {
    string timestamp = 1;
}

message FileRequest {
    string filename = 1;
    bytes content = 2;
}

message FileResponse {
    bool success = 1;
    string message = 2;
    string file_path = 3;
    int64 file_size = 4;
    string md5_hash = 5;
}

message FileContentResponse {
    bool success = 1;
    bytes content = 2;
    string message = 3;
}

message FileInfo {
    string filename = 1;
    int64 size = 2;
    string created_at = 3;
    string modified_at = 4;
    string md5_hash = 5;
}

message FilesListResponse {
    repeated FileInfo files = 1;
}